### DISCUSSION TITLE
Do fastapi support one of schema?

### DISCUSSION THREAD

**Discussion by namwooo** [fastapi/fastapi#9422] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

class PredictorName(str, Enum):
    yolov5 = "yolov5"
    SAM = "SAM"

@app.post(
    "/{predictor_name}/predict",
)
def predict(predictor_name: PredictorName, request: PredictRequestBbox):
    ...
```

### Description

Hello, folks :)

Do fastAPI support one of schema?
In example code, I wanted to have different request body schema depend on `PredictorName`. 

if predictor_name is "yolov5", request schema should be `PredictRequestBbox`,
and if predictor_name is "SAM", I want to use `PredictRequestSegmentation` instead.

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.10.0

**Comment by alexedtionweb** [comment]
-- 
Alexander Silva
| mobile: +55 (11)946590100 <+55+(11)946590100>
| phone: +1 (702)3425822 <+1+(702)3425822>
| email: ***@***.***
| address: São Paulo Area, Brazil
<http://www.linkedin.com/in/alexander-farias>
<http://telegram.me/msg_alexandersilva>
IMPORTANT: The contents of this email and any attachments are confidential.
It is strictly forbidden to share any part of this message with any third
party, without a written consent of the sender. If you received this
message by mistake, please reply to this message and follow with its
deletion, so that we can ensure such a mistake does not occur in the future.
